src/diffusers/utils/dynamic_slice.py

```diff
@@ -0,0 +1,10 @@
+from torch import Tensor
+from typing import List
+
+def dynamic_slice(
+    x: Tensor,
+    starts: List[int],
+    sizes: List[int],
+) -> Tensor:
+    slicing = [slice(start, start + size) for start, size in zip(starts, sizes)]
```

Comment:
this attempts to implement [`jax.lax.dynamic_slice()`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.dynamic_slice.html), but hey is this literally just [`torch.narrow()`](https://pytorch.org/docs/stable/generated/torch.narrow.html)?

Reply:
Yeah that works also:
https://github.com/brkirch/stable-diffusion-webui/commit/b6faa7549c7bc3c7ae025bb33600bdfc3887be62

No notable performance difference that I observed, but it's probably slightly more efficient nonetheless.